spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/firestore/GcpFirestoreAutoConfiguration.java

```diff
@@ -104,32 +105,40 @@ public Firestore firestore(FirestoreOptions firestoreOptions) {
 		return firestoreOptions.getService();
 	}
 
-	@Bean
-	@ConditionalOnMissingBean
-	public FirestoreGrpc.FirestoreStub firestoreGrpcStub(
-			ManagedChannel firestoreManagedChannel) throws IOException {
-		return FirestoreGrpc.newStub(firestoreManagedChannel)
-				.withCallCredentials(MoreCallCredentials.from(this.credentialsProvider.getCredentials()));
-	}
-
-	@Bean
-	@ConditionalOnMissingBean
-	public FirestoreMappingContext firestoreMappingContext() {
-		return new FirestoreMappingContext();
-	}
-
-	@Bean
-	@ConditionalOnMissingBean
-	public FirestoreTemplate firestoreTemplate(FirestoreGrpc.FirestoreStub firestoreStub) {
-		return new FirestoreTemplate(firestoreStub, this.firestoreRootPath);
-	}
-
-	@Bean
-	@ConditionalOnMissingBean
-	public ManagedChannel firestoreManagedChannel() {
-		return ManagedChannelBuilder
-				.forTarget(this.hostPort)
-				.userAgent(USER_AGENT_HEADER_PROVIDER.getUserAgent())
-				.build();
+	/**
+	 * The Firestore reactive template and data repositories support auto-configuration.
+	 */
+	@ConditionalOnClass({ FirestoreGrpc.FirestoreStub.class })
+	@ConditionalOnMissingBean(CloudSqlJdbcInfoProvider.class)
```

Comment:
Is doing conditional on missing bean `CloudSqlJdbcInfoProvider.class` intentional? Or is there a different bean you wanted to specify?